extendeddatasquare.go

```diff
@@ -260,6 +260,52 @@ func (eds *ExtendedDataSquare) Width() uint {
 	return eds.width
 }
 
+// FlattenedEDS returns the extended data square as a flattened slice of bytes.
+func (eds *ExtendedDataSquare) FlattenedEDS() [][]byte {
+	return eds.Flattened()
+}
+
+// FlattenedODS returns the original data square as a flattened slice of bytes.
+func (eds *ExtendedDataSquare) FlattenedODS() (flattened [][]byte) {
+	flattened = make([][]byte, eds.originalDataWidth*eds.originalDataWidth)
+	for i := uint(0); i < eds.originalDataWidth; i++ {
+		row := eds.Row(i)
+		for j := uint(0); j < eds.originalDataWidth; j++ {
+			flattened[(i*eds.originalDataWidth)+j] = row[j]
+		}
+	}
+	return flattened
+}
+
+// Equals returns true if other is equal to eds.
+func (eds *ExtendedDataSquare) Equals(other *ExtendedDataSquare) bool {
+	if eds.originalDataWidth != other.originalDataWidth {
+		return false
+	}
+	if eds.codec.Name() != other.codec.Name() {
+		return false
+	}
+	if eds.chunkSize != other.chunkSize {
+		return false
+	}
+	if eds.width != other.width {
```

Comment:
Isn't equal orig width implies equal width and vice-versa?

Reply:
Yes I think so. Put another way, this check seems unnecessary because https://github.com/celestiaorg/rsmt2d/pull/241/files/530d9c382b9a8b635ea52415ebdd6bb87635a2c5#diff-22b8afa50d1e5bd7269f72c772d718320500ae2d12cb0d077dadd8eb87124965R282-R284 so I'm open to preserving or removing. I don't have a strong preference.